README.md

```diff
@@ -19,10 +20,99 @@ This is a project initiated by Mozilla to gather code coverage results on Firefo
 * [Minimum requirements](#minimum-requirements)
 * [License](#license)
 
-## Usage
 
-1. Download grcov from https://github.com/mozilla/grcov/releases or run ```cargo install grcov```
-2. Run grcov:
+## How to get a coverage report
+
+1. Download grcov from [releases](https://github.com/mozilla/grcov/releases) or run `cargo install grcov`
+
+2. Ensure that the following environment variables are set up:
+
+```sh
+export CARGO_INCREMENTAL=0
+export RUSTFLAGS="-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Zno-landing-pads"
+```
+These will ensure that things like dead code elimination do not skew the coverage.
+
+3. Build your code:
+
+`cargo build`
+
+If you look in `target/debug/deps` dir you will see `.gcno` files have appeared. These are the locations that could be covered.
+
+4. Run your tests:
+
+`cargo test`
+
+In the `target/debug/deps/` dir you will now also see `.gcda` files. These contain the hit counts on which of those locations have been reached. Both sets of files are used as inputs to `grcov`.
+ 
+5. Run grcov!
+
+Generate a html coverage report like this:
+```sh
+grcov ./target/debug/ -s . -t html --llvm --branch --ignore-not-existing -o ./target/debug/coverage/
+```
+
+You can see the report in `target/debug/coverage/index.html`.
+
+(or alterntatively with `-t lcov` grcov will output a lcov compatible coverage report that you could then feed into lcov's `genhtml` command).
+
+
+## Alternative reports
+
+grcov provides the following output types:
+
+| Output Type `-t` | Description |
+| ---            | ---         |
+| lcov (default) | lcov's INFO format that is compatible with the linux coverage project. |
+| ade            | ActiveData\-ETL format. Only useful for mozilla projects               |
+| coveralls      | Upload coverage output directly to coveralls.io (requires token parameter). | 
+| coveralls+     | Like coveralls but with function level information. |
+| files          | Output a file list of source files. |
+| covdir         | Provides coverage in a JSON format. |
+| html           | Output a directory full of html files showing in detail the covered and uncovered source.  |
+
+
+### lcov's genhtml
```

Comment:
I'd prefer to keep these sections below, where they were before

Reply:
Sorry - I've moved that section down. Is that now a good ordering?

Reply:
They aren't where they were before :P

Reply:
To be clear, I want them where they were because it makes the diff much easier to review.
We can move them before the `man grcov` section, but in a separate PR.

Reply:
When a project is compiled in release mode with `-Zprofile` there is no instrumentation. The workaround is to add a `-g` to add debug info.
So probably just a word about that.

Reply:
@calixteman great idea - have put that in.

Reply:
@marco-c I understand now. Think that's done.